### ISSUE TITLE
Building broken on Python 3.10

### ISSUE THREAD

**Issue by ongyx** [arkane-systems/genie#234]
**Windows version (build number):**
21H1 (19043.1526)

**Linux distribution:**
Arch Linux (ArchWSL)

**Describe the bug**
Trying to build master after 2.2-dev was merged, make reported the following:

```
Collecting pathlib
  Using cached pathlib-1.0.1.tar.gz (49 kB)
    < ... >
      File "/usr/lib/python3.10/zipfile.py", line 19, in <module>
        import pathlib
      File "/tmp/pip-install-12s6f0mn/pathlib_effca9fcaa954f4ea79cd44d248d1387/pathlib.py", line 10, in <module>
        from collections import Sequence
    ImportError: cannot import name 'Sequence' from 'collections' (/usr/lib/python3.10/collections/__init__.py)
    ----------------------------------------
< ... >
ERROR: Could not find a version that satisfies the requirement pathlib (from nsenter)
ERROR: No matching distribution found for pathlib
```

TL;DR nsenter on PyPI uses an old, backported pathlib which doesn't work on Python 3.10, but is already fixed in the [upstream](https://github.com/zalando/python-nsenter/commit/b7fd78fef24c456d88130c75fe734417728e97e8) repo.
Problem is, there hasn't been a PyPI update since 2016. 

**To Reproduce**
1. Run `make -C binsrc`
2. See error

**Expected behavior**
Successful build of genie.

**Additional context**
A few suggested solutions:

- Install nsenter directly from github (with `pip install git+https://github.com/zalando/python-nsenter.git`).
  This seems like the most painless way.
- Vendorise nsenter in this repo itself.
- Maintain a fork and upload to pypi under a different name.

**I confirm that I have read the ENTIRE supplied readme file and checked for relevant information on the repository wiki before raising this issue, and that if the solution to this issue is found in either location, it will be closed without further comment:**

 - [x] Yes.

**Comment by cerebrate** [comment]
Thanks for catching this. I've implemented a fix (the installation from GitHub, for now) on the dev-2.2 branch.

**Comment by ongyx** [comment]
I've also updated the PKGBUILD on the AUR now that genie builds correctly, thanks :)

**Comment by cerebrate** [comment]
Huh. I didn't know it was on the AUR; I know it's been removed a couple of times in the past as they didn't want to have WSL-specific packages there. I hope your version sticks!

**Comment by cerebrate** [comment]
Thanks for the PKGBUILD. I put together a (somewhat modified) version for inclusion in the repo (splitting the tarball build in the Makefile to eliminate needing to be explicit in the PKGBUILD about it, having it use the default branch instead of the dev branch for post-merge, and such); feel free to backport anything that seems useful.

I did notice that to get my local Arch container to build it, I have to downgrade libffi and/or install libffi7 from the AUR. Not being familiar with Arch myself, is there any way to express that requirement in the PKGBUILD dependencies?

**Comment by ongyx** [comment]
Yes, you can add it to the PKGBUILD `depends` line. But that's strange, I built genie successfully on Arch without having to downgrade libffi.
Maybe try doing a full upgrade with `pacman -Syu` before building because Arch is a rolling release, unlike Debian.
